core/src/main/java/org/elasticsearch/script/ScriptService.java

```diff
@@ -598,9 +618,9 @@ public String toString() {
         }
     }
 
-    private static String getCacheKey(ScriptEngineService scriptEngineService, String script) {
+    private static String getCacheKey(ScriptEngineService scriptEngineService, String name, String code) {
         String lang = scriptEngineService.types()[0];
-        return lang + ":" + script;
+        return lang + ":" + name + (code != null ? ":" + code : "");
```

Comment:
In the case of an inline script, won't `name` already be the entire script? So in this case the cache key would be the code duplicated twice?